zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -783,6 +785,124 @@ benchmarks! {
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Block(start_block..end_block)),
+            ).unwrap();
+        }
+
+        let range_start: MomentOf<T> = 100_000u64.saturated_into();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+        for _ in 31..64 {
+            create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Timestamp(range_start..range_end)),
+            ).unwrap();
+        }
+
+        let block_number: T::BlockNumber = Zero::zero();
+        let last_time_frame: TimeFrame = Zero::zero();
+        for i in 1..=b {
+            <MarketIdsPerOpenBlock<T>>::try_mutate(block_number, |ids| {
+                ids.try_push(i.into())
+            }).unwrap();
+        }
+
+        let last_offset: TimeFrame = last_time_frame + 1.saturated_into::<u64>();
+        //* quadratic complexity should not be allowed in substrate blockchains
+        //* assume at first that the last time frame is one block before the current time frame
+        let t = 0;
+        let current_time_frame: TimeFrame = last_offset + t.saturated_into::<u64>();
+        for i in 1..=f {
+            <MarketIdsPerOpenTimeFrame<T>>::try_mutate(current_time_frame, |ids| {
+                // + 31 to not conflict with the markets of MarketIdsPerOpenBlock
+                ids.try_push((i + 31).into())
+            }).unwrap();
+        }
+    }: {
+        Pallet::<T>::market_status_manager::<
+            _,
+            MarketIdsPerOpenBlock<T>,
+            MarketIdsPerOpenTimeFrame<T>,
+        >(
+            block_number,
+            last_time_frame,
+            current_time_frame,
+            |market_id, market| {
+                // noop, because weight is already measured somewhere else
+                Ok(())
+            },
+        )
+        .unwrap();
+    }
+
+    market_resolution_manager {
+        let r in 1..31;
+        let d in 1..31;
+
+        let range_start: MomentOf<T> = 100_000u64.saturated_into();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+        // ensure markets exist
+        for _ in 0..64 {
+            let (_, market_id) = create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Timestamp(range_start..range_end)),
+            )?;
+            // ensure market is reported
+            T::MarketCommons::mutate_market(&market_id, |market| {
+                market.status = MarketStatus::Reported;
+                Ok(())
+            })?;
+        }
+
+        let block_number: T::BlockNumber = Zero::zero();
+
+        let mut r_ids_vec = Vec::new();
+        for i in 1..=r {
+           r_ids_vec.push(i.into());
+        }
```

Comment:
[nitpick^2] I'd use the patterns from `d_ids_vec` here, as well (or vice versa - both are fine!)